### PR TITLE
implement revert logic by using originalSelector

### PR DESCRIPTION
**Pull request by cescoferraro**
This PR add revert logic to individual assertions

## Changes

- implement field originalSelector on assertions

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [x] added a test

### CODE REVIEW

File: web/src/components/AssertionCard/AssertionCardSelectorList.tsx [hunk 1]
```diff
@@ -18,6 +19,7 @@ const AssertionCardSelectorList: React.FC<IProps> = ({selectorList, pseudoSelect
           <S.SelectorValueText>{value}</S.SelectorValueText>
         </S.Selector>
       ))}
+      <S.SelectorValueText>{originalSelector}</S.SelectorValueText>
       {pseudoSelector && (
```
Comment (cescoferraro) [inline web/src/components/AssertionCard/AssertionCardSelectorList.tsx:23]: only showing for debugging proposes

File: web/src/models/AssertionResults.model.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ const AssertionResults = ({allPassed = false, results = []}: TRawAssertionResult
       id: uniqueId(),
       selector,
       spanIds: AssertionService.getSpanIds(resultList),
+      originalSelector: selector,
```
Comment (cescoferraro) [inline web/src/models/AssertionResults.model.ts:14]: creating the field on AsserionResult

File: web/src/redux/slices/TestDefinition.slice.ts [hunk 3]
```diff
@@ -30,11 +30,31 @@ export const assertionResultsToDefinitionList = (assertionResults: TAssertionRes
...
+    removeDefinition: CaseReducer<ITestDefinitionState, PayloadAction<{selector: string}>>;
+    revertDefinition: CaseReducer<ITestDefinitionState, PayloadAction<{originalSelector?: string}>>;
+    resetDefinitionList: CaseReducer<ITestDefinitionState>;
+    setAssertionResults: CaseReducer<ITestDefinitionState, PayloadAction<TAssertionResults>>;
+    clearAffectedSpans: CaseReducer<ITestDefinitionState>;
+    setAffectedSpans: CaseReducer<ITestDefinitionState, PayloadAction<string[]>>;
+    setSelectedAssertion: CaseReducer<ITestDefinitionState, PayloadAction<string>>;
+  },
+  'testDefinition'
+>({
```
Comment (cescoferraro) [inline web/src/redux/slices/TestDefinition.slice.ts:57]: add better typescript typings

File: web/src/redux/slices/TestDefinition.slice.ts [hunk 4]
```diff
@@ -51,12 +71,33 @@ const testDefinitionSlice = createSlice({
...
       state,
       {payload: {definition, selector}}: PayloadAction<{definition: TTestDefinitionEntry; selector: string}>
     ) {
       state.definitionList = state.definitionList.map(def => {
-        if (def.selector === selector) return definition;
+        if (def.originalSelector === selector)
+          return {
+            ...def,
+            ...definition,
+            originalSelector: def.originalSelector,
```
Comment (cescoferraro) [inline web/src/redux/slices/TestDefinition.slice.ts:99]: need to ensure originalDefinition is carried over

File: web/src/components/AssertionCard/AssertionCard.tsx [hunk 5]
```diff
@@ -57,10 +66,23 @@ const AssertionCard: React.FC<TAssertionCardProps> = ({
...
+            selectorList={selectorList}
+            pseudoSelector={pseudoSelector}
+          />
         </div>
         <S.ActionsContainer>
-          {isDraft && <S.StatusTag>draft</S.StatusTag>}
+          {isDraft && (
+            <>
+              <S.StatusTag>draft</S.StatusTag>
+              <Tooltip color="white" title="Revert Assertion">
```
Comment (xoscar) [inline web/src/components/AssertionCard/AssertionCard.tsx:79]: It is kind of weird that both the tooltip an the actual message have the same text

File: web/src/redux/slices/TestDefinition.slice.ts (hunk 3 above)
Comment (xoscar) [reply web/src/redux/slices/TestDefinition.slice.ts:57]: I'm always up for this. Should this mean that we can drop the types from each of the reducer functions as well?

File: web/src/models/AssertionResults.model.ts (hunk 2 above)
Comment (jorgeepc) [reply web/src/models/AssertionResults.model.ts:14]: Do we want this in the model? I see we also have a `assertionResultsToDefinitionList` in `web/src/redux/slices/TestDefinition.slice.ts`

Comment (xoscar) [reply web/src/models/AssertionResults.model.ts:14]: I think it is good here for the moment, but I get your point

File: web/src/redux/slices/TestDefinition.slice.ts [hunk 6]
```diff
@@ -30,38 +30,70 @@ export const assertionResultsToDefinitionList = (assertionResults: TAssertionRes
...
     },
-    updateDefinition(
-      state,
-      {payload: {definition, selector}}: PayloadAction<{definition: TTestDefinitionEntry; selector: string}>
-    ) {
+    revertDefinition(state, {payload: {originalSelector}}) {
+      // @ts-ignore
+      state.definitionList = state.definitionList
+        .map(d => {
+          if (d?.originalSelector === originalSelector) {
```
Comment (xoscar) [inline web/src/redux/slices/TestDefinition.slice.ts:78]: Does this cover the scenario of reverting a newly created assertion?

**Pull request merged**